frontend/src/users/contexts/UserProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, useContext, useEffect } from 'react';
+import { useLocalStorage } from 'react-use';
+import { userAbortController } from '@users/api';
+import { JWT } from '@users/consts/localstorage';
+import { isTokenExpired } from '@users/helpers/jwt';
+import { useSubscribe } from '@users/hooks/api';
+
+interface UserState {
+  loginUser: (jwt: string | undefined) => void;
+  logoutUser: () => void;
+  isLoggedIn: boolean;
+}
+type UserProviderProps = {
+  children: React.ReactNode;
+};
+
+const UserContext = createContext<UserState>(null as any);
+
+export const useUser = (): UserState => {
+  return useContext(UserContext);
+};
+
+export const UserProvider: React.FC<UserProviderProps> = ({ children }: UserProviderProps) => {
+  const [jwt, setJwt, removeJwt] = useLocalStorage<string>(JWT);
+  useSubscribe(jwt);
+
+  const isLoggedIn = !!jwt;
+  const loginUser = (jwt: string | undefined): void => {
+    setJwt(jwt);
+  };
+  const logoutUser = (): void => {
+    removeJwt();
+    userAbortController.abort();
```

Comment:
Hmm, wydaje mi sie, ze skoro subskrybujesz w `useSubscribe`, to tez tam powinines odsubskrybowywac